mailbox/api/src/main/java/org/apache/james/mailbox/model/MailboxQuery.java

```diff
@@ -249,34 +239,33 @@ public boolean isPathMatch(MailboxPath mailboxPath) {
      *         notnull
      */
     public String getCombinedName() {
-        final String result;
-        if (base != null && base.getName() != null && base.getName().length() > 0) {
-            final int baseLength = base.getName().length();
-            if (base.getName().charAt(baseLength - 1) == pathDelimiter) {
+        String baseName = this.baseName.orElse(null);
+        if (baseName != null && baseName.length() > 0) {
```

Comment:
if baseName.isPresent

then you extract baseName

Reply:
can you provide a fix?